062-quotas-management.md

```diff
@@ -0,0 +1,163 @@
+The custom implementations can have different goals.
+For example the [Strimzi Quota plugin](https://github.com/strimzi/kafka-quotas-plugin) focus on preventing the Kafka brokers from running out of disk space instead of limiting the individual clients or users.
+
+### Quotas in Strimzi
+
+Currently, Strimzi does not have any API support for quotas in the `Kafka` custom resource.
+All quotas are configured directly by the users in `.spec.kafka.config` section of the `Kafka` custom resource.
+That gives users a lot of freedom and flexibility to configure the quotas.
+But it also does not provide them any support when the quota plugin requires some more complicated configuration.
+
+The only place where Strimzi directly supports configuring quotas is in the `KafkaUser` resource.
+It can be used to configure a specific quota for a given user using Kafka's built-in quota plugin.
+
+## Motivation
+
+The absence of an dedicated API for quota management in the `Kafka` CR currently creates two major limitations.
+1. The built-in Apache Kafka quota plugin has an ability to configure _default quotas_.
+   Default quotas apply to any user that has no specific quota set.
+   However, the default quotas cannot be configured declaratively in the `.spec.kafka.config` section.
+   And they cannot be configured in a `KafkaUser` resource either because they do not belong to any particular user.
+   The only way to configure them right now is imperatively use the Kafka Admin API.
+   While it can be used by users, it does not follow Strimzi's basic idea of declarative management.
+   Having a dedicated API for quota configuration as part of the `Kafka` CR would make it much easier to configure the default quotas.
+2. The initial releases of the Strimzi Quotas Plugin were very simple.
+   It helps to prevent users producing directly to a broker that was running out of disk space.
+   But it fails to prevent replication of messages produced to other brokers in the Kafka cluster from filling in the disk.
+   It also doesn't support many additional features such as JBOD storage.
+   Because the initial releases were very simple, they had also very simple configuration.
+   And it was possible to configure them in the `.spec.kafka.config` section of the `Kafka` CR.
+   The next - not yet released - version of the Strimzi Quotas plugin brings many improvements.
+   More details about how the new version works can be found in the [Cluster Wide Volume Usage Quota Management](./047-cluster-wide-volume-usage-quota-management.md) proposal.
+   It operates at the level of a whole Kafka cluster and it has improved ability to prevent replicated messages to fill in the broker storage.
+   It also supports JBOD storage.
+   But it has also much more complicated configuration.
+   The plugin instances running in the different Kafka brokers communicate with each other using a Kafka topic.
+   And need therefore to connect to the Kafka brokers as clients as well.
```

Comment:
The new version of the quota plug-in doesn't store any state in Kafka, in fact there is no inter node co-ordination. It does however still connect to the admin API to call `describeCluster` and `describeLogDirs`. 

That doesn't really change the motivation/goals, but is probably relevant for context.